translate/src/context/TranslationContext.tsx

```diff
@@ -1,100 +1,84 @@
-import React, {
-  createContext,
-  useContext,
-  useState,
-  Dispatch,
-  SetStateAction,
-} from 'react';
-import { fetchGPTTransform } from '~/api/machinery';
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { type MachineryTranslation, fetchGPTTransform } from '~/api/machinery';
+import { useActiveTranslation } from './EntityView';
 
-interface LLMTranslationContextType {
-  llmTranslation: string;
-  setLlmTranslation: Dispatch<SetStateAction<string>>;
+type SelState = {
   loading: boolean;
   selectedOption: string;
-  setSelectedOption: Dispatch<SetStateAction<string>>;
+  llmTranslation: string;
+};
+
+interface LLMTranslationContextType {
+  getSelState(mt: MachineryTranslation): SelState;
   transformLLMTranslation: (
-    original: string,
-    current: string,
+    mt: MachineryTranslation,
     characteristic: string,
     localeName: string,
   ) => Promise<void>;
 }
 
-const LLMTranslationContext = createContext<LLMTranslationContextType>({
-  llmTranslation: '',
-  setLlmTranslation: () => {},
+const initSelState = () => ({
   loading: false,
   selectedOption: '',
-  setSelectedOption: () => {},
+  llmTranslation: '',
+});
+
+const LLMTranslationContext = createContext<LLMTranslationContextType>({
+  getSelState: initSelState,
   transformLLMTranslation: async () => {},
 });
 
 export const LLMTranslationProvider: React.FC = ({ children }) => {
-  const [llmTranslation, setLlmTranslation] = useState<string>('');
-  const [loading, setLoading] = useState<boolean>(false);
-  const [selectedOption, setSelectedOption] = useState<string>('');
-
-  const handleSetSelectedOption = (option: string) => {
-    let displayText = '';
+  const translation = useActiveTranslation();
+  const [state, setState] = useState(new Map<MachineryTranslation, SelState>());
 
-    switch (option) {
-      case 'alternative':
-        displayText = 'REPHRASED';
-        break;
-      case 'formal':
-        displayText = 'FORMAL';
-        break;
-      case 'informal':
-        displayText = 'INFORMAL';
-        break;
-      case 'original':
-        displayText = '';
-        break;
-    }
```

Comment:
This is more appropriate to be in the component, where it'll also be easier to refactor later for proper localization. I think there was a code comment on that which has been dropped?

Reply:
Thanks for catching that, I have added the comment back.